### PR TITLE
Delete video endpoint

### PR DESCRIPTION
**Pull request by LuisiitoDev**
### Checklist

- [ ] Created/Updated Automated Test(s) for the related controller(s)
- [ ] Verified all Automated Tests are passing locally.
- [ ] Added reviewer(s)
- [ ] Referenced the related Issue/Task

### Description
It was created the endpoint for deleting a video
it was created the delete method inside VideoService

## Related Issue
https://github.com/efonsecab/FairPlayTube/issues/38

### CODE REVIEW

File: src/FairPlayTubeSln/FairPlayTube.Controllers/VideoController.cs [hunk 1]
```diff
@@ -42,6 +42,20 @@ public VideoController(VideoService videoService, IMapper mapper, ICurrentUserPr
...
+        /// </summary>
+        /// <param name="videoId"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        [HttpPost("[action]")]
+        [Authorize(Roles = Common.Global.Constants.Roles.User)]
+        public async Task DeleteVideo(string videoId, CancellationToken cancellationToken)
+        {
+            var userObjectId = this.CurrentUserProvider.GetObjectId();
+            await VideoService.DeleteVideoAsync(videoId, userObjectId, cancellationToken);
```
Comment (efonsecab) [inline src/FairPlayTubeSln/FairPlayTube.Controllers/VideoController.cs:56]: @LuisiitoDev 
Needs to validate if the user is the video's owner.
Check the logic on the other endpoints.
If the user is not an owner of the video throw an exception with the following message
"Delete denied. You are not an owner of this video".

**Review by efonsecab — CHANGES_REQUESTED** [review]

File: src/FairPlayTubeSln/FairPlayTube.Controllers/VideoController.cs (hunk 1 above)
Comment (LuisiitoDev) [reply src/FairPlayTubeSln/FairPlayTube.Controllers/VideoController.cs:56]: gotcha!, I'll do it and update the PR, thanks.

**Review by efonsecab — CHANGES_REQUESTED** [review]

File: src/FairPlayTubeSln/FairPlayTube.Controllers/VideoController.cs (hunk 1 above)
Comment (LuisiitoDev) [reply src/FairPlayTubeSln/FairPlayTube.Controllers/VideoController.cs:56]: Sure, I'm gonna do it, thanks

Comment (LuisiitoDev) [reply src/FairPlayTubeSln/FairPlayTube.Controllers/VideoController.cs]: Done, I have applied the changes required.

File: src/FairPlayTubeSln/FairPlayTube.Services/VideoService.cs [hunk 2]
```diff
@@ -55,6 +55,46 @@ TextAnalysisServices textAnalysisService
...
+        public async Task<bool> DeleteVideoAsync(string userAzureAdB2cObjectId, string videoId, CancellationToken cancellationToken)
+        {
+            var videoEntity = await this.FairplaytubeDatabaseContext.VideoInfo
+                                        .Include(p => p.ApplicationUser)
+                                        .SingleOrDefaultAsync(p => p.VideoId == videoId);
+
+            if (videoEntity == null)
+                throw new Exception($"Video: {videoId} does not exit");
+
+            if (!videoEntity.ApplicationUser.AzureAdB2cobjectId.ToString().Equals(userAzureAdB2cObjectId))
```
Comment (efonsecab) [inline src/FairPlayTubeSln/FairPlayTube.Services/VideoService.cs:67]: Move this logic to the controller, the VideoService must not have context knowledge in order to be reusable.
The onwership validation logic must be in the controller and use the IsVideoOwner method.
Check the "GetVideoEditAccessToken" endpoint to see an example

**Review by efonsecab — CHANGES_REQUESTED** [review]

File: src/FairPlayTubeSln/FairPlayTube.Services/VideoService.cs (hunk 2 above)
Comment (LuisiitoDev) [reply src/FairPlayTubeSln/FairPlayTube.Services/VideoService.cs]: Done.

**Review by efonsecab — APPROVED** [review]

**Pull request merged**